mantaray/received.py

```diff
@@ -316,14 +335,35 @@ def _handle_cap(server_view: views.ServerView, args: list[str]) -> None:
     subcommand = args[1]
     if subcommand == "ACK":
         acknowledged = set(args[-1].split())
+
         if "sasl" in acknowledged:
             server_view.core.send("AUTHENTICATE PLAIN")
+
+        for capability in acknowledged:
+            server_view.core.cap_list.add(capability)
+
     elif subcommand == "NAK":
         rejected = set(args[-1].split())
```

Comment:
If I understood correctly, this splitting doesn't make sense as the server NAKs or ACKs only a single capability at a time.

Reply:
@Akuli As the code is working now, acknowledged will ever only contain one capability since we're `REQ`ing one capability at a time. The for loop only adds one capability, but keeping it in a for loop will allow scaling if we later prefer to send several `REQ`s together.

I plan on improving Mantaray's capability negotiation in a new PR shortly.

The only potential issue here (which is currently not an issue) is that if we scale the functionality so that acknowledge can contain more than 1 capability, then `server_view.core.pending_cap_count -= 1` will not suffice. But I think this can be kept for now, since an improvement is coming soon, and atm, this cannot bug.

Reply:
It would be good if this explanation was a short comment, e.g. `# TODO: splitting currently unnecessary because server sends one capability at a time`